packages/components/pie-webc-core/src/mixins/rtl/rtlMixin.ts

```diff
@@ -24,8 +17,8 @@ declare class _RTLInterface {
  * allowing you to easily determine the text direction within your component.
  *
  * @function
- * @param {Constructor<LitElement>} superClass - The LitElement class to extend with RTL functionality.
- * @returns {Constructor<_RTLInterface> & T} - A new class extending both LitElement and _RTLInterface.
+ * @param {GenericConstructor<LitElement>} superClass - The LitElement class to extend with RTL functionality.
```

Comment:
Just to keep up my noob TS questions. Why change this? (and what's the difference between the two)

Reply:
The latter - it's just been moved into a separate file to reuse between Mixins we create now that we have more than one. This is an example of a [constrained mixin](https://www.typescriptlang.org/docs/handbook/mixins.html), annoyingly we have to write a type for a constructor to achieve this. The constraint here is that we only want to apply these to classes that extend `LitElement`.

You'll see in that docs link that TS do a similar thing in their example and name it `GConstructor` 😄